File: swift/huifier.swift
import Foundation
class Huifier {
    
    func huify(word: String) -> String {
        var result = word.lowercased()
        let vowels = "аеёиоуыэюя"
        let rules = [
            "а": "я",
            "о": "ё",
            "у": "ю",
            "ы": "и",
            "э": "е"
        ]
        for letter in result {
            if vowels.contains(letter) {
                let isInDictionary = rules.contains { (key, value) -> Bool in
                    return (key == String(letter)) || (value == String(letter))
                }
                if isInDictionary {
                    result = (rules[String(letter)] ?? String(letter)) + result.dropFirst()
                }
                break
            } else {
                result = String(result.dropFirst())
            }
        }
        return (result.isEmpty ? "хуй" : "ху\(result)").capitalized.trimmingCharacters(in: NSCharacterSet.whitespaces)
    }
    
}

File: dart/huifier.dart
String huify(String word) {
  var result = word.toLowerCase().trim();
  const vowels = 'аеёиоуыэюя';
  const rules = {
    'а': 'я',
    'о': 'е',
    'у': 'ю',
    'ы': 'и',
    'э': 'е',
  };
  while (!result.isEmpty) {
    var letter = result.substring(0, 1);
    if (vowels.contains(letter)) {
      if (rules.containsKey(letter)) {
        result = '${rules[letter]}${result.substring(1)}';
      }
      break;
    } else {
      result = result.substring(1);
    }
  }
  return !result.isEmpty ? 'Ху${result}' : 'Хуй';
}
